exaudfclient/base/javacontainer/script_options/parser_ctpg_script_importer.cc

```diff
@@ -43,35 +67,11 @@ void ScriptImporter::importScript(std::string & scriptCode,
                                   {
                                       return first.idx_in_source < second.idx_in_source;
                                   });
-        struct ReplacedScripts {
-            ReplacedScripts(ReplacedScripts&&) = default;
-            std::string script;
-            size_t origPos;
-            size_t origLen;
-        };
         std::vector<ReplacedScripts> replacedScripts;
         replacedScripts.reserve(optionIt->second.size());
         //In order to continue compatibility with legacy implementation we must collect import scripts in forward direction
         //but then replace in reverse direction (in order to keep consistency of positions)
-        for (const auto & option: optionIt->second) {
-            const char *importScriptCode = findImportScript(option.value);
-            std::string importScriptCodeStr;
-            if (m_importedScriptChecksums.addScript(importScriptCode) ) {
-                // Script has not been imported yet
-                // If this imported script contains %import statements
-                // they will be resolved in the next recursion.
-                ctpg_parser::options_map_t newOptions;
-                try {
-                    ExecutionGraph::OptionsLineParser::CTPG::parseOptions(importScriptCode, newOptions);
-                } catch(const ExecutionGraph::OptionParserException & ex) {
-                    Utils::rethrow(ex, "F-UDF-CL-SL-JAVA-1630");
-                }
-                importScriptCodeStr.assign(importScriptCode);
-                importScript(importScriptCodeStr, newOptions, recursionDepth + 1);
-            }
-            ReplacedScripts replacedScript = {.script = std::move(importScriptCodeStr), .origPos = option.idx_in_source, .origLen = option.size };
-            replacedScripts.push_back(std::move(replacedScript));
-        }
+        replaceScripts(optionIt->second, recursionDepth, replacedScripts);
```

Comment:
the extracted part doesn't replace the scripts, it only prepares everything, the next line replace it

Reply:
makes sense